### PR TITLE
Tech609 legacy applications

### PR DESCRIPTION
**Pull request by brankoEISI**

### CODE REVIEW

File: packages/employer-api/src/controllers/claim.controller.ts [hunk 1]
```diff
@@ -116,6 +116,54 @@ export const createClaim = async (req: any, res: express.Response) => {
...
+            return res.status(403).send("Forbidden")
+        }
+
+        const prefillFields = {
+            isLegacy: true,
+            container: {
+                businessAddress1: req.body.address,
+                employerCity: req.body.city
+            }
+        }
```
Comment (strdavis) [inline packages/employer-api/src/controllers/claim.controller.ts:135]: Should we require the postal code for validation like we do in the application forms, then prefill the postal code as well? It threw me off a little to have to enter the postal code in the form. Still works find like this though.

File: packages/employer-api/src/controllers/claim.controller.ts [hunk 2]
```diff
@@ -116,6 +116,54 @@ export const createClaim = async (req: any, res: express.Response) => {
...
+            prefillFields
+        )
+        if (createDraftResult?.id) {
+            const insertResult = await insertLegacyClaim(
+                req.body.formKey,
+                req.body.guid,
+                createDraftResult.id,
+                req.body.catchment,
+                req.body.storefront
+            )
```
Comment (strdavis) [inline packages/employer-api/src/controllers/claim.controller.ts:152]: Currently the position title and associated application ID do not get set to any value in the claim record. So is it correct that the form currently contains no information specifying the particular legacy application it's associated with? Maybe we could require that the user enter the position title and an application ID number, in addition to their address, before creating the legacy claim? I think there needs to be enough information to identify the associated application.

File: packages/employer-api/src/services/claim.service.ts [hunk 3]
```diff
@@ -88,6 +88,32 @@ export const insertClaim = async (
...
+    trx?: any
+) => {
+    const data = {
+        id,
+        form_submission_id: submissionID,
+        created_date: new Date(),
+        created_by: userGuid,
+        status: "Draft",
+        catchmentno: catchment,
+        workbc_centre: storefront
```
Comment (strdavis) [inline packages/employer-api/src/services/claim.service.ts:106]: workbc_centre needs to be set to catchmentNo-storefrontId, not storefrontId. This is causing the workbc column in the UI to always display 'Unassigned' for these claim forms.

File: packages/employer-api/src/utils/addressToCatchment.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable import/prefer-default-export */
+
+import pins from "../constants/centres.json"
```
Comment (strdavis) [inline packages/employer-api/src/utils/addressToCatchment.ts:3]: We need to remove catchments 46 and 47 from the list of options in centres.json.

**Review by strdavis — COMMENTED** [review]
New UI looks great and form creation is working for me! See comments regarding some issues I found. Also, catchment moving is currently failing with code 500 for these forms in my local environment.

File: packages/employer-api/src/controllers/claim.controller.ts (hunk 1 above)
Comment (brankoEISI) [reply packages/employer-api/src/controllers/claim.controller.ts:135]: The validation in the application forms doesn't actually use the postal code at all, which is why I left it out

File: packages/employer-api/src/controllers/claim.controller.ts (hunk 2 above)
Comment (brankoEISI) [reply packages/employer-api/src/controllers/claim.controller.ts:152]: Confirmed with Erin that this is fine.

**Review by strdavis — APPROVED** [review]
Nice! Looks good.

**Pull request merged**